staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -737,6 +737,9 @@ func deletionFinalizersForGarbageCollection(ctx context.Context, e *Store, acces
 	if !e.EnableGarbageCollection {
 		return false, []string{}
 	}
+	if accessor.GetDeletionTimestamp() != nil {
```

Comment:
Have you checked the [namespace registry](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/namespace/storage/storage.go#L128)? We probably need to add the same logic to the namespace registry.